bindings/lua/src/lib.rs

```diff
@@ -0,0 +1,164 @@
+    operator.set("delete", lua.create_function(operator_delete)?)?;
+    operator.set("is_exist", lua.create_function(operator_is_exist)?)?;
+    operator.set("close", lua.create_function(operator_close)?)?;
+    Ok(operator)
+}
+
+fn operator_is_exist<'a>(_: &'a Lua, (operator, path):  (LuaTable<'a>, String))-> LuaResult<bool> {
+    let op = operator.get::<_, ODOperator>("_operator")?;
+    let op = op.operator;
+
+    let path = path.as_str();
+    let res = op.is_exist(path);
+    match res {
+        Ok(exist) => Ok(exist),
+        Err(e) => Err(LuaError::external(e)),
+    }
+}
+
+// fn operator_stat<'a>(lua: &'a Lua, (operator, path):  (LuaTable<'a>, String))-> LuaResult<LuaTable<'a>> {
+//     let op = operator.get::<_, ODOperator>("_operator")?;
+//     let op = op.operator;
+
+//     let path = path.as_str();
+//     let res = op.stat(path);
+//     match res {
+//         Ok(meta) => {
+//             let lua_meta = lua.create_table()?;
+//             let entry = match meta.mode {
+//                 Some(e) => e,
+//                 None => return Err(LuaError::external("entry is empty")),
+//             };
+
+//             lua_meta.set("size", meta.size)?;
+//             lua_meta.set("is_dir", meta.is_dir)?;
+//             lua_meta.set("is_file", meta.is_file)?;
+//             lua_meta.set("is_symlink", meta.is_symlink)?;
+//             lua_meta.set("is_fifo", meta.is_fifo)?;
+//             for (key, value) in meta {
+//                 lua_meta.set(key, value)?;
+//             }
+//             Ok(lua_meta)
+//         },
+//         Err(e) => Err(LuaError::external(e)),
+//     }
+// }
+
+fn operator_delete<'a>(_: &'a Lua, (operator, path):  (LuaTable<'a>, String))-> LuaResult<()> {
+    let op = operator.get::<_, ODOperator>("_operator")?;
+    let op = op.operator;
+
+    let path = path.as_str();
+    let res = op.delete(path);
+    match res {
+        Ok(_) => Ok(()),
+        Err(e) => Err(LuaError::external(e)),
+    }
+}
+
+fn operator_write<'a>(_: &'a Lua, (operator, path, bytes):  (LuaTable<'a>, String, String))-> LuaResult<()> {
+    let op = operator.get::<_, ODOperator>("_operator")?;
+    let op = op.operator;
+
+    let path = path.as_str();
+    let res = op.write(path, bytes);
+    match res {
+        Ok(_) => Ok(()),
+        Err(e) => Err(LuaError::external(e)),
+    }
+}
+
+fn operator_read<'a>(lua: &'a Lua, (operator, path):  (LuaTable<'a>, String))-> LuaResult<LuaString<'a>> {
+    let op = operator.get::<_, ODOperator>("_operator")?;
+    let op = op.operator;
+
+    let path = path.as_str();
+    let data = op.read(path);
+    match data {
+        Ok(data) => Ok(lua.create_string(&data)?),
+        Err(e) => Err(LuaError::external(e)),
+    }
+}
+
+fn operator_close<'a>(_: &'a Lua, operator: LuaTable<'a>) -> LuaResult<()> {
+    operator.set("_operator", LuaNil)?;
+    Ok(())
+}
+
+
+fn hello(_: &Lua, name: String) -> LuaResult<()> {
```

Comment:
Please remove this not used function.

Reply:
Thanks, wait for me to finish my work(removing the WIP in the PR summary) before reviewing it. 😆

Reply:
Take your time and have fun!